pygsti/modelmembers/povms/__init__.py

```diff
@@ -498,14 +512,102 @@ def convert(povm, to_type, basis, ideal_povm=None, flatten_structure=False):
+                    base_dense_effects.append(dense_effect.reshape((1,len(dense_effect))))
+
+                dense_ideal_povm = _np.concatenate(base_dense_effects, axis=0)
+
+                #Collect all noisy effects
+                dense_effects = []
+                for effect in povm.values():
+                    dense_effect = effect.to_dense()
+                    dense_effects.append(dense_effect.reshape((1,len(dense_effect))))
+
+                dense_povm = _np.concatenate(dense_effects, axis=0)
+                
+                #It is often the case that there are more error generators than physical degrees of freedom in the POVM
+                #We define a function which finds linear comb. of errgens that span these degrees of freedom.
+                #This has been called "the trivial gauge", and this function is meant to avoid it
+                def calc_physical_subspace(dense_ideal_povm, epsilon = 1e-9):
+
+                    degrees_of_freedom = (dense_ideal_povm.shape[0] - 1) * dense_ideal_povm.shape[1]
+                    errgen = _LindbladErrorgen.from_error_generator(povm.state_space.dim, parameterization=to_type)
+
+                    if degrees_of_freedom > errgen.num_params:
+                        warnings.warn("POVM has more degrees of freedom than the available number of parameters, representation in this parameterization is not guaranteed")
+                    exp_errgen = _ExpErrorgenOp(errgen)
+                    
+                    num_errgens = errgen.num_params
+                    #TODO: Maybe we can use the num of params instead of number of matrix entries, as some of them are linearly dependent.
+                    #i.e E0 completely determines E1 if those are the only two povm elements (E0 + E1 = Identity)
+                    num_entries = dense_ideal_povm.shape[0]*dense_ideal_povm.shape[1]
+
+                    #Compute the jacobian with respect to the error generators. This will allow us to see which
+                    #error generators change the POVM entries
+                    J = _np.zeros((num_entries,num_errgens))
+                    
+                    for i in range(num_errgens):
+                        new_vec = _np.zeros(num_errgens)
+                        new_vec[i] = epsilon
+                        exp_errgen.from_vector(new_vec)
+                        vectorized_povm = _np.zeros(num_entries)
+                        perturbed_povm = (dense_ideal_povm @ exp_errgen.to_dense() - dense_ideal_povm)/epsilon 
+
+                        perturbed_povm_t = perturbed_povm.transpose()
+                        for j, column in enumerate(perturbed_povm_t):
+                            vectorized_povm[j*len(perturbed_povm_t[0]):(j+1)*len(perturbed_povm_t[0])] = column
+                        
+                        J[:,i] = vectorized_povm.transpose()
+
+                    _,S,V = _np.linalg.svd(J)
+                    return V[:len(S),]
```

Comment:
Three comments.
1. numpy and scipy SVD functions return the transpose of V, not V itself. So it's good naming convention to say ``Vt`` instead of ``V``. If you were dealing with complex matrices then you'd want to write ``Vh`` instead of ``V``.
2. If you only need the first "number of singular value" singular vectors, then you should call this as ``_, _, Vt = _np.linalg.svd(J, full_matrices=False)``.
3. This implementation isn't checking if any singular values in ``S`` are zero. If they are, then you probably want tor return fewer singular vectors. Consider how, in the current implementation, ``V[:len(S),]`` can be can arbitrary orthogonal matrix when ``J`` is the zero matrix.

Reply:
@juangmendoza19, can you take a look at points 2 and 3? My sense is that if ``S[i]`` is near zero then you wouldn't want to include ``Vt[i,:]`` among the returned vectors. Ping @coreyostrove in case you can provide insight on how (near-)zero singular values should be handled.

Reply:
Your intuition is correct, singular vectors associated with near-zero singular vectors shouldn't be included. It looks like Juan's most recent pair of commits accounts for this now.

Reply:
Just looked at the change. The logic is fine with me now, but I did leave a comment asking that the three or four lines be reduced to a single line.